src/Voron/Platform/Win32/Win32JournalWriter.cs

```diff
@@ -55,11 +55,22 @@ public void WritePages(long position, byte* p, int numberOfPages)
             if (Disposed)
                 throw new ObjectDisposedException("Win32JournalWriter");
 
-            if (numberOfPages > _maxNumberOfPages)
-                throw new ArgumentException(
-                    "Cannot write " + numberOfPages + " pages in a single write, the size is too large",
-                    nameof(numberOfPages));
+            while (numberOfPages > _maxNumberOfPagesPerSingleWrite)
+            {
+                WriteFile(position, p, _maxNumberOfPagesPerSingleWrite);
```

Comment:
TODO : We need to make a single call or mark the entire section until we finish all the segments writes